pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -206,17 +206,28 @@ func resolveTask(ctx context.Context, resolver remote.Resolver, name, namespace
 	return taskObj, refSource, vr, nil
 }
 
-func resolveStepAction(ctx context.Context, resolver remote.Resolver, name, namespace string, k8s kubernetes.Interface, tekton clientset.Interface) (*v1alpha1.StepAction, *v1.RefSource, error) {
+func resolveStepAction(ctx context.Context, resolver remote.Resolver, name, namespace string, k8s kubernetes.Interface, tekton clientset.Interface) (*v1beta1.StepAction, *v1.RefSource, error) {
 	obj, refSource, err := resolver.Get(ctx, "StepAction", name)
 	if err != nil {
 		return nil, nil, err
 	}
-	switch obj := obj.(type) { //nolint:gocritic
-	case *v1alpha1.StepAction:
+	switch obj := obj.(type) {
+	case *v1beta1.StepAction:
 		if err := apiserver.DryRunValidate(ctx, namespace, obj, tekton); err != nil {
 			return nil, nil, err
 		}
 		return obj, refSource, nil
+	case *v1alpha1.StepAction:
+		obj.SetDefaults(ctx)
+		if err := apiserver.DryRunValidate(ctx, namespace, obj, tekton); err != nil {
+			return nil, nil, err
+		}
+		v1BetaStepAction := v1beta1.StepAction{}
+		err := obj.ConvertTo(ctx, &v1BetaStepAction)
+		if err != nil {
+			return nil, nil, err
+		}
+		return &v1BetaStepAction, refSource, nil
```

Comment:
I think this needs coverage. Might be easy to replicate the tests for Tasks here.

Reply:
Good catch! Thanks! I missed the `TypeMetadata` here 🤦